### PR TITLE
:seedling: Add unit test on features.go

### LINKED ISSUE #1574: Unit test coverage improvement in BMO
Improve the unit test coverage by writing new test cases or completely new tests. Check the current coverage with `make unit-cover`.

Here are some suggestions on where to start. Note that some functions are harder to test than others because they may depend on access to network resources or a database. If you come across such function, skip them and aim for the low hanging fruits instead.

- [x] [features.go](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/provisioner/ironic/clients/features.go) has no tests at all and looks quite simple to test. Fixed in https://github.com/metal3-io/baremetal-operator/pull/1579
- [ ] [loadTLSConfigFromEnv](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/provisioner/ironic/factory.go#L169) is untested and should be easy to test. Fixed in https://github.com/metal3-io/baremetal-operator/pull/1589 or https://github.com/metal3-io/baremetal-operator/pull/2013
- [x] [baremetalhost_types.go](https://github.com/metal3-io/baremetal-operator/blob/main/apis/metal3.io/v1alpha1/baremetalhost_types.go) has multiple small functions without coverage that could easily be tested. Fixed in https://github.com/metal3-io/baremetal-operator/pull/1615
- [x] [bmceventsubscription_controller.go](https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/bmceventsubscription_controller.go) has no tests at all. Fixed in https://github.com/metal3-io/baremetal-operator/pull/1615
- [ ] [preprovisioningimage_controller.go](https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/preprovisioningimage_controller.go) has no tests at all. Fixed in https://github.com/metal3-io/baremetal-operator/pull/2013
- [x] [LoadAuth](https://github.com/metal3-io/baremetal-operator/blob/b5c4120dd294e94cd63fede91321f38623293ec7/pkg/provisioner/ironic/clients/auth.go#L43) is untested. Fixed in https://github.com/metal3-io/baremetal-operator/pull/1774
- [x] [client.go](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/provisioner/ironic/clients/client.go) has no tests at all. Fixed in https://github.com/metal3-io/baremetal-operator/pull/1606

### PR DESCRIPTION
**Pull request by lekaf974**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1574

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,3 +59,6 @@ config/overlays/temp
 
 # release-notes
 /releasenotes
+
+# jetbrains
+.idea/
```
Comment (elfosardo) [inline .gitignore:64]: nit: this should be in a global gitignore, not a repo specific one

**Review by elfosardo — COMMENTED** [review]
/lgtm

File: .gitignore (hunk 1 above)
Comment (lekaf974) [reply .gitignore]: Removed

File: pkg/provisioner/ironic/clients/features_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	type fields struct {
+		MaxVersion int
+	}
+	tests := []struct {
+		name    string
+		feature fields
+		want    string
+	}{
+		{
+			name: fmt.Sprintf("MaxVesrion < %d return microvestion %s", 86, baseline),
```
Comment (lentzi90) [inline pkg/provisioner/ironic/clients/features_test.go:19]: nit
```suggestion
			name: fmt.Sprintf("MaxVersion < %d return microversion %s", 86, baseline),
```

File: pkg/provisioner/ironic/clients/features_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			feature: fields{
+				MaxVersion: 50,
+			},
+			want: baseline,
+		},
+		{
+			name: fmt.Sprintf("MaxVesrion = %d return %s", 86, microVersion),
```
Comment (lentzi90) [inline pkg/provisioner/ironic/clients/features_test.go:26]:
```suggestion
			name: fmt.Sprintf("MaxVersion = %d return %s", 86, microVersion),
```

File: pkg/provisioner/ironic/clients/features_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			feature: fields{
+				MaxVersion: 86,
+			},
+			want: microVersion,
+		},
+		{
+			name: fmt.Sprintf("MaxVesrion > %d return %s", 86, microVersion),
```
Comment (lentzi90) [inline pkg/provisioner/ironic/clients/features_test.go:33]:
```suggestion
			name: fmt.Sprintf("MaxVersion > %d return %s", 86, microVersion),
```

**Review by lentzi90 — COMMENTED** [review]
Thanks for this!
Some typos below

**Review by lentzi90 — COMMENTED** [review]
Great!
One more thing, could you please squash the commits?

File: pkg/provisioner/ironic/clients/features_test.go (hunk 2 above)
Comment (lekaf974) [reply pkg/provisioner/ironic/clients/features_test.go]: Fixed

File: pkg/provisioner/ironic/clients/features_test.go (hunk 3 above)
Comment (lekaf974) [reply pkg/provisioner/ironic/clients/features_test.go]: Fixed

File: pkg/provisioner/ironic/clients/features_test.go (hunk 4 above)
Comment (lekaf974) [reply pkg/provisioner/ironic/clients/features_test.go]: Fixed

**Comment by tuminoid** [comment]
/override test-centos-e2e-integration-main 
Unit test addition, skip.

**Pull request merged**